apps/api/src/app/bridge/usecases/sync/sync.usecase.ts

```diff
@@ -142,149 +161,134 @@ export class Sync {
     });
   }
 
-  private async createWorkflows(
+  private async upsertWorkflows(
     command: SyncCommand,
     workflowsFromBridge: DiscoverWorkflowOutput[]
-  ): Promise<NotificationTemplateEntity[]> {
+  ): Promise<WorkflowResponseDto[]> {
     return Promise.all(
       workflowsFromBridge.map(async (workflow) => {
-        const workflowExist = await this.notificationTemplateRepository.findByTriggerIdentifier(
+        const fetchedWorkflow = await this.notificationTemplateRepository.findByTriggerIdentifier(
           command.environmentId,
           workflow.workflowId
         );
 
-        let savedWorkflow: NotificationTemplateEntity | undefined;
+        const upsertWorkflowCommand = fetchedWorkflow
```

Comment:
Commands should follow the use case pattern. That is, we should encapsulate the create or update logic of a command inside the upsert command or reuse the create or update upsert logic of the usecase.